File: data/multilang_comparison/program_julia/program_julia/src/func6.jl
include("program_julia_head.jl")
function func6(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array191 = func7(params0, get_path(), loopsFactor);
      array191.refC -= 1
      if array191.refC == 0
          empty!(array191.data)
          array191 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array201 = func8(params0, get_path(), loopsFactor);
      array201.refC -= 1
      if array201.refC == 0
          empty!(array201.data)
          array201 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array206 = func9(params0, get_path(), loopsFactor);
   return array206;
end

File: data/multilang_comparison/program_go/program_go/src/func6.go
package main
import "math/rand"
var _ = rand.Intn
func Func6(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array191 *Array = Func7(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array191.RefC--
      if array191.RefC == 0 {
          array191.Data = nil
          array191 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array201 *Array = Func8(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array201.RefC--
      if array201.RefC == 0 {
          array201.Data = nil
          array201 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array206 *Array = Func9(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array206;
}

